### PR TITLE
[Port From Paradise Station] Syndicate Teleporter

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

It's spelunki's hand teleporter. Teleports you forward on use. When teleported in a wall, teleports you again. If that fails, you get gibbed.

https://github.com/ParadiseSS13/Paradise/pull/14934

## Changelog
:cl:
add: Syndicate teleporter in uplinks
/:cl:

### CODE REVIEW

File: code/game/objects/items/teleportation.dm [hunk 1]
```diff
@@ -312,3 +312,154 @@
...
+	var/max_charges = 4
+	var/minimum_teleport_distance = 4
+	var/maximum_teleport_distance = 8
+	var/saving_throw_distance = 3
+	var/recharge_time = 200 //20 Seconds
+	var/recharging = FALSE
+
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] has [charges] out of [max_charges] charges left.</span>"
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:343]: `	. += span_notice("[src] has [charges] out of [max_charges] charges left.")`

File: code/game/objects/items/teleportation.dm [hunk 2]
```diff
@@ -312,3 +312,154 @@
...
+/obj/item/teleporter/proc/recharge()
+	charges++
+	recharging = FALSE
+	check_charges()
+
+/obj/item/teleporter/emp_act(severity)
+	if(prob(50 / severity))
+		if(istype(loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/user = loc
+			to_chat(user, "<span class='danger'>[src] buzzes and activates!</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:365]: `to_chat(user, span_danger("[src] buzzes and activates!"))`

File: code/game/objects/items/teleportation.dm [hunk 3]
```diff
@@ -312,3 +312,154 @@
...
+			attempt_teleport(user, TRUE) //EMP Activates a teleport with no safety.
+		else
+			visible_message("<span class='warning'>[src] activates and blinks out of existence!</span>")
+			do_sparks(2, 1, src)
+			qdel(src)
+
+/obj/item/teleporter/proc/attempt_teleport(mob/user, EMP_D = FALSE)
+	if(!charges)
+		to_chat(user, "<span class='warning'>[src] is still recharging.</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:374]: `to_chat(user, span_warning("[src] is still recharging."))`

File: code/game/objects/items/teleportation.dm [hunk 4]
```diff
@@ -312,3 +312,154 @@
...
+	check_charges()
+
+/obj/item/teleporter/emp_act(severity)
+	if(prob(50 / severity))
+		if(istype(loc, /mob/living/carbon/human))
+			var/mob/living/carbon/human/user = loc
+			to_chat(user, "<span class='danger'>[src] buzzes and activates!</span>")
+			attempt_teleport(user, TRUE) //EMP Activates a teleport with no safety.
+		else
+			visible_message("<span class='warning'>[src] activates and blinks out of existence!</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:368]: `visible_message(span_warning("[src] activates and blinks out of existence!"))`

File: code/game/objects/items/teleportation.dm [hunk 5]
```diff
@@ -312,3 +312,154 @@
...
+
+/obj/item/teleporter/proc/attempt_teleport(mob/user, EMP_D = FALSE)
+	if(!charges)
+		to_chat(user, "<span class='warning'>[src] is still recharging.</span>")
+		return
+
+	var/turf/current_location = get_turf(user)
+	var/area/current_area = current_location.loc
+	if(!current_location || !(current_area.area_flags & NOTELEPORT) || is_away_level(current_location.z) || is_centcom_level(current_location.z) || !isturf(user.loc))//If turf was not found or they're on z level 2 or >7 which does not currently exist. or if user is not located on a turf
+		to_chat(user, "<span class='notice'>\The [src] is malfunctioning.</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:380]: `to_chat(user, span_notice("\The [src] is malfunctioning."))`

File: code/game/objects/items/teleportation.dm [hunk 6]
```diff
@@ -312,3 +312,154 @@
...
+
+/obj/item/teleporter/proc/get_fragged(mob/user, turf/destination)
+	var/turf/mobloc = get_turf(user)
+	user.forceMove(destination)
+	playsound(mobloc, "sparks", 50, TRUE)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+	playsound(destination, "sparks", 50, TRUE)
+	playsound(destination, "sound/magic/disintegrate.ogg", 50, TRUE)
+	to_chat(user, "<span class='userdanger'>You teleport into the wall, the teleporter tries to save you, but-</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:428]: `to_chat(user, span_userdanger("You teleport into the wall, the teleporter tries to save you, but-"))`

File: code/game/objects/items/teleportation.dm [hunk 7]
```diff
@@ -312,3 +312,154 @@
...
+	destination.ex_act(2) //Destroy the wall
+	if (iscarbon(user))
+		var/mob/living/carbon/cuser = user
+		cuser.gib(FALSE,FALSE,FALSE,FALSE)
+
+/obj/item/teleporter/proc/telefrag(turf/fragging_location, mob/user)
+	for(var/mob/living/M in fragging_location)//Hit everything in the turf
+		M.apply_damage(20, BRUTE)
+		M.Paralyze(30)
+		to_chat(M, "<span class='userdanger'>[user] teleports into you, knocking you to the floor with the bluespace wave!</span>")
```
Comment (MACIEKBAKI) [inline code/game/objects/items/teleportation.dm:438]: `to_chat(M, span_userdanger("[user] teleports into you, knocking you to the floor with the bluespace wave!"))`

**Review by MACIEKBAKI — CHANGES_REQUESTED** [review]
just replace all `"<span class='CLASSNAME'>TEXT HERE</span>"` with `span_CLASSNAME("TEXT HERE")` and I think it'll be good

**Comment by MACIEKBAKI** [comment]
If you want a very simple and lazy way to replace the spans with span defines, go to a regex testing site like [regex101](url), go into substitution, use the regular expression `"<span class='(.+)'>(.+)<\/span>"` and replacement string `span_$1("$2")` and paste the code into the test string, then just copy the output and it should work without major issues.

**Comment by eeSPee** [comment]
Thanks, @MACIEKBAKI

Should be fixed now.

**Review by MACIEKBAKI — APPROVED** [review]

**Pull request merged**